core/src/main/java/io/substrait/type/Type.java

```diff
@@ -11,9 +11,6 @@
 public interface Type extends TypeExpression, ParameterizedType, NullableType, FunctionArg {
   static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(Type.class);
 
-  public static final TypeCreator REQUIRED = TypeCreator.REQUIRED;
-  public static final TypeCreator NULLABLE = TypeCreator.NULLABLE;
```

Comment:
In tests, these fields would consistently end up as null. See failures in:
https://github.com/substrait-io/substrait-java/actions/runs/6203665019/job/16844629387

This started happening once I added additional code to the TypeCreator, which makes me suspect some weird initialisation race conditions.

IntellIJ was also hinted me in that direction:
```
Modifier 'static' is redundant for interface fields
```

Moving to using TypeCreator.REQUIRED and TypeCreator.NULLABLE seems to have resolved the issue.

Reply:
Talking through this with a co-worker, `TypeCreator` references `Type` and `Type` references `TypeCreator`, which may also affect the initialization.